services/src/modules/authentication/strategies/jwt.ts

```diff
@@ -0,0 +1,25 @@
+import fastify from 'fastify';
+import { decode } from 'jsonwebtoken';
+import logger from '../../logger';
+import { authenticationConfig } from '../../config';
+
+export default async function (request: fastify.FastifyRequest): Promise<void> {
+  const authHeader: string = request.headers.authorization;
+  if (!authHeader) return Promise.reject(new Error('Unauthorized'));
+  const token = authHeader.split(' ')[1];
+  const decodedToken = decode(token, { json: true }) as Record<string, unknown>;
+  const issuer = String(decodedToken?.iss);
+  const issuerConfig = authenticationConfig?.jwt?.[issuer];
+  if (!issuerConfig) {
+    logger.warn({ issuer }, 'Unknown issuer');
+    return Promise.reject(new Error('Unauthorized'));
+  }
+  if (issuerConfig.authenticatedPaths.some(ap => request.raw.url?.endsWith(ap))) {
+    try {
+      await request.jwtVerify();
+    } catch (error) {
+      logger.warn(error, 'Failed to verify request JWT');
+      return Promise.reject(new Error('Unauthorized'));
```

Comment:
It will automatically wrap it in a rejected Promise since we are in an `async` function

```suggestion
      throw new Error('Unauthorized');
```